src/GafferImage/FormatQuery.cpp

```diff
@@ -0,0 +1,200 @@
+{
+	float cv;
+
+	if( & child == parent.getChild( 0 ) )
+	{
+		cv = value.x;
+	}
+	else if( & child == parent.getChild( 1 ) )
+	{
+		cv = value.y;
+	}
+	else
+	{
+		assert( 0 ); // NOTE : Unknown child plug
+		cv = 0.f;
+	}
+
+	child.setValue( cv );
+
+}
+
+} // namespace
+
+namespace GafferImage
+{
+
+size_t FormatQuery::g_firstPlugIndex = 0;
+
+GAFFER_NODE_DEFINE_TYPE( FormatQuery );
+
+FormatQuery::FormatQuery( std::string const& name )
+: Gaffer::ComputeNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new ImagePlug( "image" ) );
+	addChild( new GafferImage::FormatPlug( "format", Gaffer::Plug::Out ) );
+	addChild( new Gaffer::V2fPlug( "center", Gaffer::Plug::Out ) );
+	addChild( new Gaffer::V2iPlug( "size", Gaffer::Plug::Out ) );
+}
+
+FormatQuery::~FormatQuery()
+{}
+
+ImagePlug* FormatQuery::imagePlug()
+{
+	return const_cast< ImagePlug* >(
+		static_cast< FormatQuery const* >( this )->imagePlug() );
+}
+
+ImagePlug const* FormatQuery::imagePlug() const
+{
+	return getChild< ImagePlug >( g_firstPlugIndex );
+}
+
+FormatPlug* FormatQuery::formatPlug()
+{
+	return const_cast< FormatPlug* >( static_cast< FormatQuery const* >( this )->formatPlug() );
+}
+
+FormatPlug const* FormatQuery::formatPlug() const
+{
+	return getChild< FormatPlug >( g_firstPlugIndex + 1 );
+}
+
+Gaffer::V2fPlug* FormatQuery::centerPlug()
+{
+	return const_cast< Gaffer::V2fPlug* >(
+		static_cast< FormatQuery const* >( this )->centerPlug() );
+}
+
+Gaffer::V2fPlug const* FormatQuery::centerPlug() const
+{
+	return getChild< Gaffer::V2fPlug >( g_firstPlugIndex + 2 );
+}
+
+Gaffer::V2iPlug* FormatQuery::sizePlug()
+{
+	return const_cast< Gaffer::V2iPlug* >(
+		static_cast< FormatQuery const* >( this )->sizePlug() );
+}
+
+Gaffer::V2iPlug const* FormatQuery::sizePlug() const
+{
+	return getChild< Gaffer::V2iPlug >( g_firstPlugIndex + 3 );
+}
+
+void FormatQuery::affects( Gaffer::Plug const* const input, AffectedPlugsContainer& outputs ) const
+{
+	ComputeNode::affects( input, outputs );
+
+	if( input == imagePlug()->formatPlug() )
+	{
+		outputs.push_back( formatPlug()->displayWindowPlug()->minPlug()->getChild( 0 ) );
+		outputs.push_back( formatPlug()->displayWindowPlug()->minPlug()->getChild( 1 ) );
+		outputs.push_back( formatPlug()->displayWindowPlug()->maxPlug()->getChild( 0 ) );
+		outputs.push_back( formatPlug()->displayWindowPlug()->maxPlug()->getChild( 1 ) );
+		outputs.push_back( formatPlug()->pixelAspectPlug() );
+		outputs.push_back( centerPlug()->getChild( 0 ) );
+		outputs.push_back( centerPlug()->getChild( 1 ) );
+		outputs.push_back( sizePlug()->getChild( 0 ) );
+		outputs.push_back( sizePlug()->getChild( 1 ) );
+	}
+}
+
+void FormatQuery::hash( Gaffer::ValuePlug const* const output, Gaffer::Context const* const context, IECore::MurmurHash& h ) const
+{
+	ComputeNode::hash( output, context, h );
+	imagePlug()->formatPlug()->hash( h );
```

Comment:
We should be using `imagePlug()->formatHash()` here, so that we get the benefit of the GlobalScope it creates. I suggest we also add a test case that would expose any context leakage - perhaps driving `Checkerboard.size` and leaving the rest to the built-in ContextSanitiser would be sufficient?